core/database/api/acl_router.js

```diff
@@ -0,0 +1,450 @@
+'use strict';
+
+const   createRouter = require('@arangodb/foxx/router');
+const   router = createRouter();
+const   joi = require('joi');
+const   g_db = require('@arangodb').db;
+const   g_lib = require('./support');
+
+module.exports = router;
+
+
+//==================== ACL API FUNCTIONS
+
+router.post('/update', function (req, res) {
+    try {
+        var result = [];
+
+        g_db._executeTransaction({
+            collections: {
+                read: ["u","p","uuid","accn","d","c","a","admin","alias"],
+                write: ["c","d","acl"]
+            },
+            action: function() {
+                const client = g_lib.getUserFromClientID( req.body.client );
+                var object = g_lib.getObject( req.body.id, client );
+                var owner_id = g_db.owner.firstExample({ _from: object._id })._to;
+                //var owner = g_db._document( owner_id );
+                //owner_id = owner_id.substr(2);
+
+                //console.log("obj:",object);
+
+                var is_coll;
+
+                if ( object._id[0] == "c" )
+                    is_coll = true;
+                else
+                    is_coll = false;
+
+                if ( !is_coll && object._id[0] != "d" )
+                    throw [g_lib.ERR_INVALID_PARAM,"Invalid object type, "+object._id];
+
+                var is_admin = true;
+
+                if ( !g_lib.hasAdminPermObject( client, object._id )){
+                    is_admin = false;
+                    if ( !g_lib.hasPermissions( client, object, g_lib.PERM_SHARE ))
+                        throw g_lib.ERR_PERM_DENIED;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/ORNL/DataFed2/pullRequest?prid=8791729)
```suggestion
                        {throw g_lib.ERR_PERM_DENIED;}
```